### PR TITLE
feat: Added support for deny rules

### PR DESCRIPTION
**Pull request by dgonzalez**
Added support for Org deny rules.

### CODE REVIEW

**Comment by BartoszBlizniak** [comment]
LGMT  @dgonzalez - thinking ahead we should probably create a `data_source` that would read the current deny rules as well and documentation for both.

File: cloudsmith/resource_deny_rule.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+				Type:         schema.TypeString,
+				Description:  "The slug of the deny rule.",
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Description:  "Description of the rule.",
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
```
Comment (dkatavic) [inline cloudsmith/resource_deny_rule.go:176]: I think that description can be empty

File: cloudsmith/resource_deny_rule.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+			"package_query": {
+				Type:         schema.TypeString,
+				Description:  "The query to match the packages to be blocked.",
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Description: "Is the rule enabled?.",
+				Optional:    true,
```
Comment (dkatavic) [inline cloudsmith/resource_deny_rule.go:187]: should `enabled` maybe be required and we can set default value?

**Comment by dkatavic** [comment]
How much effort would it be into creating a test? It would be useful to make sure that it works properly as we evolve deny rune API

**Comment by dgonzalez** [comment]
@dkatavic I am adding tests. Just testing before going deeper... and Also description can be optional yes. Good catch.

**Review by dkatavic — APPROVED** [review]
preemptively approving

File: cloudsmith/resource_deny_rule.go (hunk 2 above)
Comment (dgonzalez) [reply cloudsmith/resource_deny_rule.go]: If it is required, has to be specified by the user in the resource. e.g.:
```
resource "cloudsmith_package_deny_policy" "test-2" {
  name = "test-me-tf-2-2"
  description = "test"
  enabled = true
  package_query = "bananas"
  namespace = "cloudsmith-test"
}
```
So I think we want it to be optional with default `true` so we can do:
```
resource "cloudsmith_package_deny_policy" "test-2" {
  name = "test-me-tf-2-2"
  description = "test"
  package_query = "bananas"
  namespace = "cloudsmith-test"
}
```
And defaults to `true`. Does it make sense?

**Comment by dgonzalez** [comment]
@BartoszBlizniak the `data_source` has been created. Missing tests that are coming in a subsequent PR.

File: cloudsmith/resource_deny_rule.go (hunk 2 above)
Comment (dkatavic) [reply cloudsmith/resource_deny_rule.go]: Makes sense

**Pull request merged**